frontend/src/components/global-dialogs/motd-modal/cached-motd-modal.tsx

```diff
@@ -0,0 +1,43 @@
+'use client'
+/*
+ * SPDX-FileCopyrightText: 2023 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import React, { useCallback, useMemo, useState } from 'react'
+import { useMotdContextValue } from '../../motd/motd-context'
+import { useLocalStorage } from 'react-use'
+import { MOTD_LOCAL_STORAGE_KEY } from './fetch-motd'
+import { MotdModal } from './motd-modal'
+
+/**
+ * Reads the motd from the context and shows it in a modal.
+ * If the modal gets dismissed by the user then the "last modified" identifier will be written into the local storage
+ * to prevent that the motd will be shown again until it gets changed.
+ */
+export const CachedMotdModal: React.FC = () => {
+  const contextValue = useMotdContextValue()
+  const [cachedLastModified, saveLocalStorage] = useLocalStorage<string>(MOTD_LOCAL_STORAGE_KEY, undefined)
+
+  const [dismissed, setDismissed] = useState(false)
+
+  const show = useMemo(() => {
+    const lastModified = contextValue?.lastModified
+    return cachedLastModified !== lastModified && lastModified !== undefined && !dismissed
+  }, [cachedLastModified, contextValue?.lastModified, dismissed])
+
+  const doDismiss = useCallback(() => {
+    const lastModified = contextValue?.lastModified
+    if (lastModified) {
+      saveLocalStorage(lastModified)
+    }
+    setDismissed(true)
+  }, [contextValue, saveLocalStorage])
+
+  if (contextValue?.lastModified === undefined && process.env.NODE_ENV === 'test') {
+    return <span data-testid='loaded not visible' />
```

Comment:
Wouldn't it be better to use the `testId` method in case that changes over time?

Reply:
That right. Changed it